### DISCUSSION TITLE
Custom elements/tags

### DISCUSSION THREAD

**Discussion by neerajKumarGaianSolutions** [dita-ot/dita-ot#4512] [Q&A]
I would like to create custom DITA tags. Can someone guide me on how to achieve this?

**Comment by raducoravu** [comment]
These would be the only tutorials I know on the subject: https://dita4practitioners.github.io/dita-specialization-tutorials/standalone-tutorials/config-and-extend-tutorials-standalone.html

**Reply by neerajKumarGaianSolutions** [reply]
Thank you for your assistance. Can you confirm if it's possible to create custom elements in DITA-OT by developing plugins? I've included a sample DITA file featuring a custom tag, <custom-element>, and its corresponding XSLT file for HTML processing. Additionally, there are several predefined DITA tags, like 'shortdesc', that aren't valid HTML tags. Could you recommend resources where I can learn how these tags are transpiled into HTML?

example.dita

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE topic PUBLIC "-//OASIS//DTD DITA Topic//EN" "topic.dtd">
<topic id="example-topic">
  <title>Example Topic with Custom Element</title>
  <body>
    <p>This is a standard paragraph in DITA.</p>
    <foreign>
      <custom-element>Custom content goes here.</custom-element>
    </foreign>
  </body>
</topic>
```

XSLT file:

```xml
<xsl:template match="foreign/custom-element">
  <div class="custom-element">
    <xsl:value-of select="."/>
  </div>
</xsl:template>
```

**Comment by raducoravu** [comment]
Sometime ago I created a DITA OT plugin which allows embedding HTML content inside the DITA foreign element:

https://github.com/oxygenxml/dita-embed-html-xslfo

As in the example I gave in the readme I used a CDATA to wrap the entire HTML content so that I did not have to define a DITA XML specialization for it. And the XSLT stylesheets of the plugin just pass the entire CDATA fragment to the HTML output.

**Reply by neerajKumarGaianSolutions** [reply]
Thank you for your reply. I would like to understand how DITA-OT converts the 'shortdesc' tag into a paragraph (<p> tag) when processing a DITA file to HTML.

**Comment by stefan-jung** [comment]
Probably it's worth reading this chapter in the specification: https://www.oxygenxml.com/dita/1.3/specs/archSpec/base/coding-requirements.html
